File: src/main/java/org/acg/user_services/service/UserService.java
package org.acg.user_services.service;

import org.acg.user_services.repository.entity.UserInfo;
import org.springframework.security.core.userdetails.User;
import org.springframework.security.core.userdetails.UserDetails;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

public interface UserService {

    void save(UserInfo userInfo);
    UserInfo findByEmail(String email);
    void createPasswordResetTokenForUser(UserInfo userInfo, String token);
    void changeUserPassword(UserInfo userInfo, String password);

}

File: src/main/kotlin/org/acg/user/service/UserService.kt
package org.acg.user.service;

import org.acg.user.repository.entity.UserInfo;

interface UserService {

    fun save(userInfo: UserInfo)
    fun findByEmail(email: String): UserInfo?
    fun createPasswordResetTokenForUser(userInfo: UserInfo, token: String)
    fun changeUserPassword(userInfo: UserInfo, password: String)

}
